crates/iota-indexer/src/test_utils.rs

```diff
@@ -42,64 +42,51 @@ impl IndexerTypeConfig {
 }
 
 pub async fn start_test_indexer(
-    db_url: Option<String>,
+    db_url: String,
+    drop_and_create_db: bool,
+    reset_db: bool,
     rpc_url: String,
     reader_writer_config: IndexerTypeConfig,
     data_ingestion_path: Option<PathBuf>,
-    new_database: Option<&str>,
 ) -> (PgIndexerStore, JoinHandle<Result<(), IndexerError>>) {
     start_test_indexer_impl(
         db_url,
+        drop_and_create_db,
+        reset_db,
         rpc_url,
         reader_writer_config,
-        // reset_database
-        false,
         data_ingestion_path,
         CancellationToken::new(),
-        new_database,
     )
     .await
 }
 
 /// Starts an indexer reader or writer for testing depending on the
-/// `reader_writer_config`. If `reset_database` is true, the database instance
-/// named in `db_url` will be dropped and reinstantiated.
+/// `reader_writer_config`.
 pub async fn start_test_indexer_impl(
-    db_url: Option<String>,
+    db_url: String,
```

Comment:
This now requires that the `db_name` is present in the URL, whereas this was not the case previously I think . 

At any case, it makes sense to ensure that the db-name is included.

Reply:
Yes, I share your view! `db_url` should really point to the db.
Previously, if `new_database` was `Some` we added the name to the provided url or replaced the db name in the default URL using:

```
    if let Some(new_database) = new_database {
        db_url = replace_db_name(&db_url, new_database).0;
        ....
    };
```

It seems like in all of these `Some` cases however only name replacements happened (as the URL was interpolated by `get_indexer_db_url`) or the URL came from a default implementation.

So, I guess there is no further action to do here at this point. In future I think it would make sense to have proper type for the database url for more safety.